dashboards-observability/public/components/visualizations/charts/data_table/data_table.tsx

```diff
@@ -46,6 +33,30 @@ export const DataTable = ({ visualizations, layout, config }: any) => {
     metadata: { fields = [] },
   } = visualizations.data.rawVizData;
 
+  const { dataConfig = {} } = visualizations?.data?.userConfigs;
+  const enablePagination =
+    typeof dataConfig?.chartStyles?.enablePagination !== 'undefined'
+      ? dataConfig?.chartStyles?.enablePagination
+      : visualizations.vis.enablepagination;
+
+  const showTableHeader =
+    typeof dataConfig?.chartStyles?.showTableHeader !== 'undefined'
+      ? dataConfig?.chartStyles?.showTableHeader
+      : visualizations.vis.showtableheader;
+
+  const colunmFilter =
+    typeof dataConfig?.chartStyles?.colunmFilter !== 'undefined'
+      ? dataConfig?.chartStyles?.colunmFilter
+      : visualizations.vis.colunmfilter;
+
+  const columnAlignment =
+    dataConfig?.chartStyles?.columnAlignment || visualizations.vis.columnalignment;
+
+  const columnWidth =
+    typeof dataConfig?.chartStyles?.columnWidth !== 'undefined'
+      ? dataConfig?.chartStyles?.columnWidth
+      : visualizations.vis.columnwidth;
+
   useEffect(() => {
     document.addEventListener('keydown', hideGridFullScreenHandler);
     return () => {
```

Comment:
from Line 70, seems like we can also use useMemo hook here, could we add this for column calculation?